### DISCUSSION TITLE
Scaling Teleport Kubernetes Agents

### DISCUSSION THREAD

**Discussion by alwaysastudent** [gravitational/teleport#11209] [Q&A] [answered]
Hi, Is it necessary, or even possible, to scale the Teleport Kubernetes agents horizontally? 
I mean specifically about the Teleport agents run with just `kubernetes_service` as enabled, and not as a proxy or auth services. 

I'm assuming that the Teleport Kubernetes agents are stateless and only require an ephemeral pod level empty dir because logs and session data are shipped to the auth service via TCP, so does it require any shared storage when multiple replicas are run? Or is it even accurate to run multiple replicas of Teleport K8s agents?

**Comment by webvictim** [comment, marked as answer]
You can definitely run multiple replicas of Kubernetes agents if you want, to provide redundancy across multiple AZs/worker nodes. You can even set `highAvailability.replicaCount` and `highAvailability.requireAntiAffinity` in the chart values to automatically do this for you. 

Teleport should round-robin requests across multiple agents which provide connectivity to the same cluster. The agents are indeed stateless and don't require local storage.
